src/cromshell/alias/command.py

```diff
@@ -0,0 +1,107 @@
+import logging
+
+import click
+import csv
+import fileinput
+
+from cromshell.utilities import workflow_id_utils
+
+
+LOGGER = logging.getLogger(__name__)
+
+
+@click.command(name="alias")
+@click.argument("workflow_id", required=True)
+@click.argument("alias", required=True)
+@click.pass_obj
+def main(config, workflow_id: str, alias: str):
+    """
+    Label the given workflow ID with the given alias.
+    Aliases can be used in place of workflow IDs to reference jobs."
+    """
+
+    LOGGER.info("alias")
+
+    # Perform Checks
+    run_alias_pre_checks(
+        alias_name=alias,
+        workflow_id=workflow_id,
+        submission_file_path=config.submission_file_path
+    )
+
+    # Set workflow id with given alias
+    set_alias_for_workflow_id(
+        alias_name=alias,
+        workflow_id=workflow_id,
+        submission_file_path=config.submission_file_path
+    )
+
+    return 0
+
+
+def run_alias_pre_checks(alias_name: str, workflow_id: str, submission_file_path: str):
+    """Do several checks with input confirm it fine to create the alias"""
+
+    # check if provided alias contains white spaces or start with a dash
+    if not alias_is_valid(alias_name):
+        LOGGER.error(
+            "Alias %s is invalid, it may not start with a dash or contain whitespace.",
+            alias_name
+        )
+        raise ValueError(
+            f"Alias {alias_name} is invalid, it may not start with a dash or contain "
+            f"whitespace. "
+        )
+
+    # check if alias already exists
+    if alias_exists(
+            alias_name=alias_name, submission_file=submission_file_path
+    ):
+        LOGGER.error("Can't use %s as alias, as it is already being used.", alias_name)
+        raise ValueError(f"Alias already exists: {alias_name} ")
+
+    # check if workflow id exists
+    if not workflow_id_utils.workflow_id_exists(
+            workflow_id=workflow_id,
+            submission_file=submission_file_path
+    ):
+        LOGGER.error("Could not find workflow id %s in submission file.", workflow_id)
+        raise ValueError(
+            f"Could not find workflow id {workflow_id} in submission file.")
+
+
+def alias_is_valid(alias_name: str) -> bool:
+    """Check if alias name starts with '-' or has a whitespace char"""
+    if alias_name.startswith("-") or " " in alias_name:
+        return False
+    else:
+        return True
+
+
+def alias_exists(alias_name: str, submission_file) -> bool:
+    """Check if alias name already exists in submission file"""
+    with open(submission_file, 'r') as csv_file:
+        reader = csv.DictReader(csv_file, delimiter="\t")
+        for row in reader:
+            if row["ALIAS"] == alias_name:
+                return True
+        return False
+
+
+def set_alias_for_workflow_id(
```

Comment:
There is a similar [function](https://github.com/broadinstitute/cromshell/blob/ae051d9c630fedc988b31f2d1a41160524d659ab/src/cromshell/status/command.py#L102) in the status command that updates the status for a row, I'm wondering if these two functions should be combined or separate?

Reply:
consolidate

Reply:
done